packages/@aws-cdk/aws-lambda/test/test.vpc-lambda.ts

```diff
@@ -228,6 +244,58 @@ export = {
       });
     });
 
+    test.doesNotThrow(() => {
+      new lambda.Function(stack, 'PrivateLambda', {
+        code: new lambda.InlineCode('foo'),
+        handler: 'index.handler',
+        runtime: lambda.Runtime.NODEJS_10_X,
+        vpc,
+        vpcSubnets: { subnetType: ec2.SubnetType.PRIVATE },
+      });
+    });
+
+    test.doesNotThrow(() => {
+      new lambda.Function(stack, 'IsolatedLambda', {
+        code: new lambda.InlineCode('foo'),
+        handler: 'index.handler',
+        runtime: lambda.Runtime.NODEJS_10_X,
+        vpc,
+        vpcSubnets: { subnetType: ec2.SubnetType.ISOLATED },
+      });
+    });
+    test.done();
+  },
+
+  'picking public subnet type is not allowed if not overriding allowPublicSubnet'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'VPC', {
+      subnetConfiguration: [
+        {
+          name: 'Public',
+          subnetType: ec2.SubnetType.PUBLIC,
+        },
+        {
+          name: 'Private',
+          subnetType: ec2.SubnetType.PRIVATE,
+        },
+        {
+          name: 'Isolated',
+          subnetType: ec2.SubnetType.ISOLATED,
+        },
+      ],
+    });
+
+    // WHEN
+    test.throws(() => {
+      new lambda.Function(stack, 'PublicLambda', {
+        code: new lambda.InlineCode('foo'),
+        handler: 'index.handler',
+        runtime: lambda.Runtime.NODEJS_10_X,
+        vpc,
+        vpcSubnets: { subnetType: ec2.SubnetType.PUBLIC },
+      });
+    });
```

Comment:
Check that the error is the one that we actually expect, and not some other validation/error.

```suggestion
    }, /Lambda Functions in a public subnet/);
```